pshtt/pshtt.py

```diff
@@ -654,88 +685,51 @@ def https_check(endpoint):
         logging.exception("{}: Unknown exception examining trust.".format(endpoint.url))
         utils.debug("{}: Unknown exception examining trust: {}".format(endpoint.url, err))
 
-    try:
-        cert_response = cert_plugin_result.as_text()
-    except AttributeError:
-        logging.exception("{}: Known error in sslyze 1.X with EC public keys. See https://github.com/nabla-c0d3/sslyze/issues/215".format(endpoint.url))
-        return None
-    except Exception as err:
-        endpoint.unknown_error = True
-        logging.exception("{}: Unknown exception in cert plugin.".format(endpoint.url))
-        utils.debug("{}: {}".format(endpoint.url, err))
-        return
-
-    # Debugging
-    # for msg in cert_response:
-    #     print(msg)
-
     # Default endpoint assessments to False until proven True.
     endpoint.https_expired_cert = False
     endpoint.https_self_signed_cert = False
     endpoint.https_bad_chain = False
     endpoint.https_bad_hostname = False
 
-    # STORE will be either "Mozilla" or "Custom"
-    # depending on what the user chose.
-
-    # A certificate can have multiple issues.
-    for msg in cert_response:
-
-        # Check for missing SAN.
-        if (
-            (("DNS Subject Alternative Names") in msg) and
-            (("[]") in msg)
-        ):
-            endpoint.https_bad_hostname = True
-
-        # Check for certificate expiration.
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (("certificate has expired") in msg)
-        ):
-            endpoint.https_expired_cert = True
+    cert_chain = cert_plugin_result.certificate_deployments[0].received_certificate_chain
+    leaf_cert = cert_chain[0]
+
+    # Check for leaf certificate expiration/self-signature.
+    if leaf_cert.not_valid_after < datetime.datetime.now():
+        endpoint.https_expired_cert = True
+
+    # Check to see if the cert is self-signed
+    if leaf_cert.issuer is leaf_cert.subject:
+        endpoint.https_self_signed_cert = True
+
+    # Check certificate chain
+    # NOTE: If this is the only flag that's set, it's probably
+    # an incomplete chain
+    # If this isnt the only flag that is set, it's might be
```

Comment:
One small fix in a comment as a suggested change:

```suggestion
    # If this isn't the only flag that is set, it might be
```

Reply:
Fixed in 81882c488e099cd96e1525e62e275cf7759f506f